src/liboslexec/batched_llvm_gen.cpp

```diff
@@ -1263,6 +1263,7 @@ LLVMGEN(llvm_gen_add)
     OSL_ASSERT(!A.typespec().is_array() && !B.typespec().is_array());
     if (Result.typespec().is_closure()) {
         OSL_ASSERT(A.typespec().is_closure() && B.typespec().is_closure());
+        OSL_ASSERT( !A.is_uniform() && !B.is_uniform() && !result_is_uniform );
```

Comment:
Good assert to verify our assumption that any closures will implicitly be varying